File: backend/src/main/java/es/codeurjc/backend/dto/UserDTO.java
package es.codeurjc.backend.dto;

import java.util.List;

public record UserDTO(    
    Long id,
    String fullName,
    String userName,
    Integer phone,
    String email,
    String password,
    Integer age,
    Integer numTicketsBought,
    String favoriteGenre,
    Boolean image,
    List<TicketDTO> tickets,
    List<String> roles
) {}

File: frontend/src/app/dtos/user.dto.ts
import { TicketDTO } from "./ticket.dto";

export interface UserDTO {
    id?: number,
    fullName: string,
    userName: string,
    phone: number,
    email: string,
    password: string,
    age: number,
    numTicketsBought: number,
    favoriteGenre: string,
    image: boolean,
    roles: string[],
    tickets: TicketDTO[],
    csrfToken?: string
}